packages/rollup-contracts/test/rollup-list/CanonicalTransactionChain.spec.ts

```diff
@@ -243,25 +262,83 @@ describe('CanonicalTransactionChain', () => {
           .appendTransactionBatch(DEFAULT_BATCH, localBatch.timestamp)
       })
 
-      it('should revert when appending a block with a newer timestamp', async () => {
-        const newTimestamp = localBatch.timestamp + 1
+      it('should revert when there is an older batch in the L1ToL2Queue', async () => {
+        const snapshotID = await provider.send('evm_snapshot', [])
+        await provider.send('evm_increaseTime', [FORCE_INCLUSION_PERIOD])
+        const newTimestamp = localBatch.timestamp + 60
         await TestUtils.assertRevertsAsync(
-          'Cannot submit a batch with a timestamp in the future',
+          'Must process older L1ToL2Queue batches first to enforce timestamp monotonicity',
           async () => {
             await canonicalTxChain
               .connect(sequencer)
               .appendTransactionBatch(DEFAULT_BATCH, newTimestamp)
           }
         )
+        await provider.send('evm_revert', [snapshotID])
+      })
+    })
+
+    describe('when there is a batch in the SafetyQueue', async () => {
+      let localBatch
+      beforeEach(async () => {
+        localBatch = await enqueueAndGenerateSafetyBatch(DEFAULT_TX)
+      })
+
+      it('should succesfully append a batch with an older timestamp', async () => {
+        const oldTimestamp = localBatch.timestamp - 1
+        await canonicalTxChain
+          .connect(sequencer)
+          .appendTransactionBatch(DEFAULT_BATCH, oldTimestamp)
+      })
+
+      it('should succesfully append a batch with an equal timestamp', async () => {
+        await canonicalTxChain
+          .connect(sequencer)
+          .appendTransactionBatch(DEFAULT_BATCH, localBatch.timestamp)
+      })
+
+      it('should revert when there is an older batch in the SafetyQueue', async () => {
+        const snapshotID = await provider.send('evm_snapshot', [])
+        await provider.send('evm_increaseTime', [FORCE_INCLUSION_PERIOD])
+        const newTimestamp = localBatch.timestamp + 60
+        await TestUtils.assertRevertsAsync(
+          'Must process older SafetyQueue batches first to enforce timestamp monotonicity',
+          async () => {
+            await canonicalTxChain
+              .connect(sequencer)
+              .appendTransactionBatch(DEFAULT_BATCH, newTimestamp)
+          }
+        )
+        await provider.send('evm_revert', [snapshotID])
+      })
+    })
+    describe('when both the safetyQueue and l1ToL2Queue are not empty', async () => {
```

Comment:
Might be worth adding tests to this section which mirror the `it('should revert when there is an older batch in the SafetyQueue'` test above--I think ideally we'd cover cases where there is an older batch in the safety queue, in the L1ToL2 queue, or both.

Reply:
fixed 576c035586f50313bbd6a898ad307d6b2487faaf